src/condor_utils/nvidia_utils.cpp

```diff
@@ -0,0 +1,167 @@
+/***************************************************************
+ *
+ * Copyright (C) 1990-2024, Condor Team, Computer Sciences Department,
+ * University of Wisconsin-Madison, WI.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ * 
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+#ifdef LINUX
+
+#include "condor_common.h"
+#include <tuple>
+#include <vector>
+#include <string>
+#include <cstdint>
+#include <filesystem>
+#include <charconv>
+#include <stdio.h>
+#include <cstring>
+
+#include "stl_string_utils.h"
+#include "condor_debug.h"
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/sysmacros.h>
+
+using uuidMap = std::vector<std::string, dev_t>;
+
+// Return a single std::string with the contents of a file path
+// or empty string if can't read the file
+static std::string
+file_to_string(const std::filesystem::path &p) {
+	std::string contents;
+	FILE *f = fopen(p.c_str(), "r");
+	if (f) {
+		char buf[128];
+		while (fgets(buf, sizeof(buf) - 1,f)) {
+			contents += buf;
+		}
+		fclose(f);
+	} 
+	return contents;
+}
```

Comment:
Is this different an important way from `readShortFile()` other than not using the `safe_*()`, which it should maybe not be doing?

Reply:
Yes -- `readShortFile` assumes that `stat(2)` shows the length of the file, which is not the case for files in /proc or /dev.  The stat filesize of these guys will always return a length of 0.

Reply:
Could you add a comment to that effect?  ("This works with files in /proc and /dev, but readShortFile() doesn't, because ...")  Thanks.